opm/parser/eclipse/EclipseState/Schedule/Schedule.cpp

```diff
@@ -1118,13 +1118,12 @@ namespace Opm {
 
 
     void Schedule::handleCOMPDAT( const DeckKeyword& keyword, size_t currentStep) {
-        std::map<std::string , std::vector< CompletionPtr> > completionMapList = Completion::completionsFromCOMPDATKeyword( keyword );
-        std::map<std::string , std::vector< CompletionPtr> >::iterator iter;
+        const auto wells = this->getWells( currentStep );
+        auto completions = Completion::fromCOMPDAT( keyword, wells );
+
+        for( const auto pair : completions )
```

Comment:
`getWells` returns a vector by value.

I'd like us to start returning proxies with iterators for things like this, but we're not quite there yet unfortunately.